packages/@aws-cdk/aws-apigateway/test/test.domains.ts

```diff
@@ -330,6 +330,58 @@ export = {
       }
     }));
 
+    test.done();
+  },
+
+  '"addBasePathMapping" can be used to add base path mapping to the domain with specific stage'(test: Test) {
+    // GIVEN
+    const stack = new Stack();
+    const api1 = new apigw.RestApi(stack, 'api1');
+    const api2 = new apigw.RestApi(stack, 'api2');
+    const domain = new apigw.DomainName(stack, 'my-domain', {
+      domainName: 'example.com',
+      certificate: acm.Certificate.fromCertificateArn(stack, 'cert', 'arn:aws:acm:us-east-1:1111111:certificate/11-3336f1-44483d-adc7-9cd375c5169d'),
+      endpointType: apigw.EndpointType.REGIONAL
+    });
+    api1.root.addMethod('GET');
+    api2.root.addMethod('GET');
+
+    const testDeploy = new apigw.Deployment(stack, 'test-deployment', {
+      api: api1
+    });
+
+    const testStage = new apigw.Stage(stack, 'test-stage', {
+      deployment : testDeploy
+    });
+
+    // WHEN
+    domain.addBasePathMapping(api1, { basePath: 'api1', stage: testStage });
+    domain.addBasePathMapping(api2, { basePath: 'api2' });
+
+    // THEN
+    expect(stack).to(haveResource('AWS::ApiGateway::BasePathMapping', {
+      "DomainName": {
+        "Ref": "mydomain592C948B"
+      },
+      "BasePath": "api1",
+      "RestApiId": {
+        "Ref": "api1A91238E2"
+      },
+      "Stage": { Ref: stack.getLogicalId(testStage.node.findChild('Resource') as CfnElement) }
```

Comment:
This should work I believe -

```suggestion
      "Stage": stack.resolve(testStage.stageName)
```

can you check?